writer.go

```diff
@@ -65,18 +69,23 @@ func (p *MasterPlaylist) Append(uri string, chunklist *MediaPlaylist, params Var
 		version(&p.ver, 4) // so it is optional and in theory may be set to ver.1
 		// but more tests required
 	}
-	p.buf.Reset()
+	p.dirty = true
 }
 
 func (p *MasterPlaylist) ResetCache() {
-	p.buf.Reset()
+	p.lock.Lock()
+	p.dirty = true
+	p.lock.Unlock()
```

Comment:
For code consistency, please `p.lock.Lock()` followed by `defer p.lock.Unlock()`

Reply:
fixed